src/components/ArticlePage/index.js

```diff
@@ -44,12 +45,38 @@ export class ArticlePage extends React.Component {
     this.handleBookmarkClick = this.handleBookmarkClick.bind(this);
     this.handleCommentLikeClick = this.handleCommentLikeClick.bind(this);
     this.handleCommentDislikeClick = this.handleCommentDislikeClick.bind(this);
+    this.identifyUserBookmarks = this.identifyUserBookmarks.bind(this);
+    this.updateBookmarkStatus = this.updateBookmarkStatus.bind(this);
   }
 
   componentDidMount() {
-    const { fetchSingleArticle, match } = this.props;
+    const {
+      fetchSingleArticle, match, isLoggedIn, fetchUserBookmarkedArticles
+    } = this.props;
     fetchSingleArticle(match.params.id);
     articlePageScript();
+    if (isLoggedIn) fetchUserBookmarkedArticles();
+  }
+
+  componentDidUpdate(prevProps) {
+    // const { bookmarkState } = this.state;
+    const { bookmarks, isLoggedIn } = this.props;
+    const { bookmarkStatus, bookmarkedArticles } = bookmarks;
+    if ((prevProps.isLoggedIn !== isLoggedIn && !isLoggedIn)) {
+      return this.updateBookmarkStatus(false);
+    }
+    if ((prevProps.bookmarks.bookmarkStatus !== bookmarkStatus) && bookmarkStatus) {
+      toastr.success('you have added this article to your bookmarks');
+      return this.updateBookmarkStatus(true);
+    }
+    if ((prevProps.bookmarks.bookmarkStatus !== bookmarkStatus) && !bookmarkStatus) {
+      toastr.success('you have unbookmarked this article');
```

Comment:
I feel all these logic should happen on click event of the bookmark icon

Reply:
@sulenchy. No. The click event of the bookmark icon sends a request to the server either to bookmark or un-bookmark an article. I do need to teach my component to listen for failure or success response from the server.  So I do the listening here, on the componentDidUpate lifecycle method. In this way, I am able to communicate to the user if their quest to bookmark an article was successful or not.